### ISSUE TITLE
Hyperfine not recognising LC_ALL=C or globs

### ISSUE THREAD

**Issue by Kaos-Industries** [sharkdp/hyperfine#90]
I'm using this relatively complicated `for` loop combined with `hyperfine` to compare the relative speeds of different versions of `grep`. 

```
for i in {1..15}; \
do
(hyperfine -m 10 --show-output -p 'sync64 -nobanner' 'LC_ALL=C grep -i "ajndoandajskaskaksnaodnasnakdnaosnaond" *.txt') 2>&1 |
tee -a "../Tests/grep Test 5 (Hyperfine Full Directory w. Sync).txt"; \
(hyperfine -m 10 --show-output -p 'sync64 -nobanner' 'LC_ALL=C grep -Fi "ajndoandajskaskaksnaodnasnakdnaosnaond" *.txt') 2>&1 |
tee -a "../Tests/grep -F Test 5 (Hyperfine Full Directory w. Sync).txt"; \
(hyperfine -m 10 --show-output -p 'sync64 -nobanner' 'rg -i "ajndoandajskaskaksnaodnasnakdnaosnaond" *.txt') 2>&1 |
tee -a "../Tests/ripgrep Test 5 (Hyperfine Full Directory w. Sync).txt"; \
done;
```

However, it fails with the following output:

> Benchmark #1: LC_ALL=C grep -i "ajndoandajskaskaksnaodnasnakdnaosnaond" *.txt
> 
> Flushing: B C D
> 'LC_ALL' is not recognized as an internal or external command,
> operable program or batch file.
> Error: Command terminated with non-zero exit code. Use the '-i'/'--ignore-failure' option if you want to ignore this. Alternatively, use the '--show-output' option to debug what went wrong.
> Benchmark #1: LC_ALL=C grep -Fi "ajndoandajskaskaksnaodnasnakdnaosnaond" *.txt
> 
> Flushing: B C D
> 'LC_ALL' is not recognized as an internal or external command,
> operable program or batch file.
> Error: Command terminated with non-zero exit code. Use the '-i'/'--ignore-failure' option if you want to ignore this. Alternatively, use the '--show-output' option to debug what went wrong.
> Benchmark #1: rg -i "ajndoandajskaskaksnaodnasnakdnaosnaond" *.txt
> 
> Flushing: B C D
> *.txt: The filename, directory name, or volume label syntax is incorrect. (os error 123)
> Error: Command terminated with non-zero exit code. Use the '-i'/'--ignore-failure' option if you want to ignore this. Alternatively, use the '--show-output' option to debug what went wrong.

Each command in the loop passed through `hyperfine` fails with an error, with two of them being distinct errors here. The first two suggest that `hyperfine` doesn't support commands prefixed with the [`LC=ALL_C` environment variable](https://unix.stackexchange.com/questions/87745/what-does-lc-all-c-do), and the second appears to have trouble with the glob in `*.txt`. Does hyperfine not support these, or am I using it incorrectly?

**Comment by sharkdp** [comment]
Can you please try to find a minimal example to reproduce this? What happens if you execute the same commands in your shell?

In principle, hyperfine should be able to execute any shell command. It calls `sh -c "..."` internally.

The following works for me:
``` bash
hyperfine -m 10 'LC_ALL=C grep -i "foo" *.txt'
```

Note that `grep` will return a non-zero exit code if it doesn't find any results. In such a case, you might add ` || true` to the end of the command or use `hyperfine`s `--ignore-failure` option.

**Comment by Kaos-Industries** [comment]
I learnt that removing `--show-output` makes the command run perfectly, although I could have sworn I'd already tried this when trouble-shooting yesterday. I'm not entirely sure why --show-output isn't allowed to be in there, and whether this is a bug or not, but I don't need it for my particular purposes, I only added it for the purpose of debugging.

What I can confirm though is that the most minimal example of the command I can reproduce, with no other options from `hyperfine`, still fails with the `LC_ALL` error:

`hyperfine --show-output 'LC_ALL=C grep -i "ajndoandajskaskaksnaodnasnakdnaosnaond" *.txt'`

**Comment by Kaos-Industries** [comment]
Scratch that, it was a false positive. It appears there's something else at work here; it's possible `--show-output` wasn't causing the actual problems, but simply showing me what the problem was. When I run the above minimal version of the command and leave out `--show-output`, I just get a command that benchmarks how fast each run can fail, which is incidentally pretty damn fast. When I combine `--show-output` with `--ignore-error`, I get this, which is presumably the error output from each run:

> 'LC_ALL' is not recognized as an internal or external command,
> operable program or batch file.
> 'LC_ALL' is not recognized as an internal or external command,
> operable program or batch file.
> 'LC_ALL' is not recognized as an internal or external command,
> operable program or batch file.
> 'LC_ALL' is not recognized as an internal or external command,
> operable program or batch file.
> 'LC_ALL' is not recognized as an internal or external command,
> operable program or batch file.
> 'LC_ALL' is not recognized as an internal or external command,
> operable program or batch file.
> 'LC_ALL' is not recognized as an internal or external command,
> operable program or batch file.
> 'LC_ALL' is not recognized as an internal or external command,
> operable program or batch file.
> 'LC_ALL' is not recognized as an internal or external command,
> operable program or batch file.
> 'LC_ALL' is not recognized as an internal or external command,
> operable program or batch file.
> 'LC_ALL' is not recognized as an internal or external command,
> operable program or batch file.
> 'LC_ALL' is not recognized as an internal or external command,
> operable program or batch file.
> 'LC_ALL' is not recognized as an internal or external command,
> operable program or batch file.
> 'LC_ALL' is not recognized as an internal or external command,
> operable program or batch file.
> 'LC_ALL' is not recognized as an internal or external command,
> operable program or batch file.
> 'LC_ALL' is not recognized as an internal or external command,
> operable program or batch file.
> 'LC_ALL' is not recognized as an internal or external command,
> operable program or batch file.
> 'LC_ALL' is not recognized as an internal or external command,
> operable program or batch file.
> 'LC_ALL' is not recognized as an internal or external command,
> operable program or batch file.
> 'LC_ALL' is not recognized as an internal or external command,
> operable program or batch file.
> 'LC_ALL' is not recognized as an internal or external command,
> operable program or batch file.
> 'LC_ALL' is not recognized as an internal or external command,
> operable program or batch file.
> 'LC_ALL' is not recognized as an internal or external command,
> operable program or batch file.

Still very much lost as to what's going on here.

**Comment by sharkdp** [comment]
It works fine for me. What happens if you run the same command in the terminal?
```bash
LC_ALL=C grep -i "ajndoandajskaskaksnaodnasnakdnaosnaond" *.txt
```

**Comment by sharkdp** [comment]
What happens if you just run the `LC_ALL=C grep ..` command in the terminal?

**Comment by Kaos-Industries** [comment]
Sorry, I just saw that you meant the command without `hyperfine`. Prefixing `LC_ALL=C` without hyperfine works exactly as it should, as does using `time LC_ALL=C...`, I've been using both pretty often over the last few weeks. It's just hyperfine that I'm noticing this problem with.

**Comment by sharkdp** [comment]
Ahh.. I might have an idea. I think hyperfine will call `cmd.exe /C` instead of `sh -c` on Windows. Could you try running
```
> cmd.exe /C "LC_ALL=C grep -i "ajndoandajskaskaksnaodnasnakdnaosnaond" *.txt"
```
and see if that generates the same error message?

**Comment by Kaos-Industries** [comment]
> 
> 
> Ahh.. I might have an idea. I think hyperfine will call `cmd.exe /C` instead of `sh -c` on Windows. Could you try running
> 
> ```
> > cmd.exe /C "LC_ALL=C grep -i "ajndoandajskaskaksnaodnasnakdnaosnaond" *.txt"
> ```
> 
> and see if that generates the same error message?

That gives me a single:

> 'LC_ALL' is not recognized as an internal or external command,
operable program or batch file.

Though this time when I add `--show-output` I get:

> '--show-output' is not recognized as an internal or external command,
> operable program or batch file.

**Comment by Kaos-Industries** [comment]
Good catch though, it's only just occurred to me that that error message is CMD's own, don't know why I never noticed that before. Is there a way to make hyperfine use `sh -c` when run from a `bash` shell?

**Comment by sharkdp** [comment]
> Though this time when I add `--show-output` I get [..]

Well, you can't pass `--show-output` to `cmd.exe` :smile: 

> Is there a way to make hyperfine use `sh -c` when run from a `bash` shell?

I think we could make this possible via #61.

In the meantime, I think the following might work for you:
``` bash
> export LC_ALL=C
> hyperfine --show-output -i 'grep -i "ajndoandajskaskaksnaodnasnakdnaosnaond" *.txt'
```

(Note that `cmd.exe` doesn't understand wildcards like `*.txt`, but it might be that your version of grep does the wildcard expansion)
